lib/network_utils.ml

```diff
@@ -484,7 +484,18 @@ module Ip = struct
 
   let get_mtu dev = int_of_string (List.hd (link dev "mtu"))
 
-  let get_mac dev = List.hd (link dev "link/ether")
+  let get_mac dev =
+    match link dev "link/ether" with
+    | [] -> (
+      match link dev "link/infiniband" with
+      | m :: _ ->
+          m
+      | [] ->
+          error "can't find mac address for %s" dev ;
+          ""
```

Comment:
Is it not reasonable to error here, rather than returning `""`? Seen as we used to error with `List.hd` before (but with a sensible error message of course)